File: android/src/main/kotlin/com/ezride/flutter_bg_location_plugin/handlers/StartTrackingHandler.kt
package com.ezride.flutter_bg_location_plugin.handlers

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import android.content.Context
import android.content.Intent
import android.util.Log
import com.ezride.flutter_bg_location_plugin.services.LocationService

class StartTrackingHandler : Handler{


    override val callMethod : String ="startTracking";

    override fun handler(context: Context, call: MethodCall, result: MethodChannel.Result){
        Log.d("FlutterLocationPlugin", "startTracking invoked")
        // Получаем параметр seconds и hash из аргументов. это время для сессии трекинга.
        val args = call.arguments as? Map<*, *>
        val seconds = (args?.get("seconds") as? Int) ?: 0;
        val hash = (args?.get("hash") as? String) ?: "";
        val orderId = (args?.get("orderId") as? String) ?: "";
        val isStarted =  LocationService.startTracking(context,seconds,hash,orderId);
        result.success(isStarted);

    }
}

File: ios/Classes/Handler/StartTrackingHandler.swift
final class StartTrackingHandler: Handler {

    let callMethod = "startTracking"

    func handler(call: FlutterMethodCall, result: @escaping FlutterResult) {

        guard let args = call.arguments as? [String: Any] else { return result(false) }

        let seconds = args?["seconds"] as? Int ?? 0;
        let hash = args?["hash"] as? String ?? "";
        let orderId = args?["orderId"] as? String ?? "";
        // пример работы с UserDefaults
        if let h = args["hash"] as? String {
            UserDefaults.standard.set(h, forKey: "hash")
        }

        let svc = PluginContext.shared.service            // <-singleton
        if !svc.isRunning { svc.start() }

        result(true)
    }
}